sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -49,14 +54,30 @@ export class BatchingReceiver extends MessageReceiver {
   }
 
   /**
-   * Clear the token renewal timer and set the `detachedError` property.
+   * Clear the token renewal timer and reject promise returned by current receive request.
    * @param {AmqpError | Error} [receiverError] The receiver error if any.
    * @returns {Promise<void>} Promise<void>.
    */
   async onDetached(receiverError?: AmqpError | Error): Promise<void> {
-    // Clears the token renewal timer. Closes the link and its session if they are open.
+    if (this._newMessageReceivedTimer) {
```

Comment:
Is there any chance that the following could occur:
1. User calls receive and receives events.
2. User calls receive again.
3. onDetached from 1st receive call is fired. Clears timeouts and rejects promise for 2nd receive call (due to using instance references).
4. 2nd receive call erroneously rejects.

My intuition is that if `onDetached` is called, it's going to impact all the receive calls that are in flight (do we allow calling receive concurrently? I'm assuming we don't since we have isReceivingMessages) so the behaviour would be fine, but I wanted to make sure it was thought about.

Reply:
`receiveMessages` cannot be called concurrently, so there will be only 1 receive operation on the batching receiver in progress at any time.

I worry that there are a few more corner cases like these and so wanted to open a conversation via the draft PR.

My initial instinct was to use an indicator on the connection of whether it is open or not and use that to reject the promise from inside of the `finalAction()`

But, unfortunately, rhea's `isOpen()` on link, session and connection level all return true in this case. More investigation is needed